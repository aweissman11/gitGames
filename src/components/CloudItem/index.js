import React from 'react';

export const CloudItem = (props) => (
    <div { ...props } className="tag-item-wrapper">
        <div>
            { props.text }
        </div>
        <div className="tag-item-tooltip">
            {props.text}: {props.instances}
        </div>
    </div>
);

export default CloudItem;