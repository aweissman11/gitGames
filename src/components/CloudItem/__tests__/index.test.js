import React from 'react';
import { shallow } from 'enzyme';

import CloudItem from '../index';

describe('CloudItem', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<CloudItem text='word' instances={90} />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  
})
