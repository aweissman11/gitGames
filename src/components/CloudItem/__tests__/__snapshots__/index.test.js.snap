// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloudItem should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CloudItem
    instances={90}
    text="word"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          word
        </div>,
        <div
          className="tag-item-tooltip"
        >
          word
          : 
          90
        </div>,
      ],
      "className": "tag-item-wrapper",
      "instances": 90,
      "text": "word",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "word",
        },
        "ref": null,
        "rendered": "word",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "word",
            ": ",
            90,
          ],
          "className": "tag-item-tooltip",
        },
        "ref": null,
        "rendered": Array [
          "word",
          ": ",
          90,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            word
          </div>,
          <div
            className="tag-item-tooltip"
          >
            word
            : 
            90
          </div>,
        ],
        "className": "tag-item-wrapper",
        "instances": 90,
        "text": "word",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "word",
          },
          "ref": null,
          "rendered": "word",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "word",
              ": ",
              90,
            ],
            "className": "tag-item-tooltip",
          },
          "ref": null,
          "rendered": Array [
            "word",
            ": ",
            90,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
